### PR TITLE
fix: improve ecies error fatal variants

### LINKED ISSUE #11259: all ecies error are treated as fatal
### Describe the feature

discovered during #11244 and #11254

https://github.com/paradigmxyz/reth/blob/2e05ec008a85576c146afb8325cfc48e1785da10/crates/net/network/src/error.rs#L217-L217

these include various variants that are not "fatal"

https://github.com/paradigmxyz/reth/blob/2e05ec008a85576c146afb8325cfc48e1785da10/crates/net/ecies/src/error.rs#L26-L87

## TODO
we need more detailed matching here and only treat hard crypto errors as fatal.

see https://github.com/paradigmxyz/reth/blob/2e05ec008a85576c146afb8325cfc48e1785da10/crates/net/network/src/error.rs#L123-L123

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by caglaryucekaya**
closes #11259 

Which error variants should be fatal needs to be reviewed.

### CODE REVIEW

File: crates/net/network/src/error.rs [hunk 1]
```diff
@@ -214,7 +215,19 @@ impl SessionError for PendingSessionHandshakeError {
...
+                    ECIESErrorImpl::TagCheckHeaderFailed |
+                    ECIESErrorImpl::TagCheckBodyFailed |
+                    ECIESErrorImpl::InvalidAuthData |
+                    ECIESErrorImpl::InvalidAckData |
+                    ECIESErrorImpl::InvalidHeader |
+                    ECIESErrorImpl::Secp256k1(_) |
+                    ECIESErrorImpl::InvalidHandshake { .. } |
+                    ECIESErrorImpl::UnreadableStream |
+                    ECIESErrorImpl::StreamTimeout
+            ),
```
Comment (Rjected) [inline crates/net/network/src/error.rs:230]: could we convert this into a fn on `ECIESErrorImpl`?

File: crates/net/network/src/error.rs [hunk 2]
```diff
@@ -214,7 +215,19 @@ impl SessionError for PendingSessionHandshakeError {
...
+                ECIESErrorImpl::TagCheckDecryptFailed |
+                    ECIESErrorImpl::TagCheckHeaderFailed |
+                    ECIESErrorImpl::TagCheckBodyFailed |
+                    ECIESErrorImpl::InvalidAuthData |
+                    ECIESErrorImpl::InvalidAckData |
+                    ECIESErrorImpl::InvalidHeader |
+                    ECIESErrorImpl::Secp256k1(_) |
+                    ECIESErrorImpl::InvalidHandshake { .. } |
+                    ECIESErrorImpl::UnreadableStream |
+                    ECIESErrorImpl::StreamTimeout
```
Comment (Rjected) [inline crates/net/network/src/error.rs:229]: hmm, I'm thinking `StreamTimeout` should not be fatal?

**Review by Rjected — DISMISSED** [review]
A few suggestions, but this is a great start!

File: crates/net/network/src/error.rs (hunk 2 above)
Comment (mattsse) [inline crates/net/network/src/error.rs:229]: these should not be fatal

File: crates/net/network/src/error.rs (hunk 1 above)
Comment (mattsse) [reply crates/net/network/src/error.rs:230]: we only need this here, so this is fine

File: crates/net/network/src/error.rs [hunk 3]
```diff
@@ -214,7 +215,19 @@ impl SessionError for PendingSessionHandshakeError {
     fn is_fatal_protocol_error(&self) -> bool {
         match self {
             Self::Eth(eth) => eth.is_fatal_protocol_error(),
-            Self::Ecies(_) => true,
+            Self::Ecies(err) => matches!(
```
Comment (mattsse) [inline crates/net/network/src/error.rs:218]: we also need the same match in `merits_discovery_ban`:

https://github.com/paradigmxyz/reth/blob/main/crates/net/network/src/error.rs#L210-L210

**Review by mattsse — CHANGES_REQUESTED** [review]
ty,
almost there

File: crates/net/network/src/error.rs (hunk 3 above)
Comment (caglaryucekaya) [reply crates/net/network/src/error.rs:228]: Done

**Review by mattsse — APPROVED** [review]

**Pull request merged**